contribute/design/cstor-operator/day2ops.md

```diff
@@ -0,0 +1,309 @@
+        capacity: 12345234
+
+  - hostName: gke-cstor-it-default-pool-2
+
+    groups:
+    - groupName: group1
+      blockDevices:
+      - blockDeviceName: pool-2-bd-1
+        // System generated
+        id: pool-2-bd1
+        // System generated
+        capacity: 12345234
+
+      - blockDeviceName: pool-2-bd-2
+        // System generated
+        id: pool-2-bd2
+        // System generated
+        capacity: 12345234
+
+```
+
+Expanding mirrored pools -- the spec will look
+following:
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-mirrored
+spec:
+  type: mirrored
+  nodes:
+
+  - hostName: gke-cstor-it-default-pool-1
+    groups:
+    - groupName: group1
+      blockDevices:
+      - blockDeviceName: pool-1-bd-1
+        id: pool-1-bd1
+        // System generated
+        capacity: 12345234
+
+      - blockDeviceName: pool-1-bd-2
+        id: pool-1-bd2
+        // System generated
+        capacity: 12345234
+
+    // New block device group added
+    - groupName: group2
+      blockDevices:
+      - blockDeviceName: pool-1-bd-3
+        id: pool-1-bd3
+        // System generated
+        capacity: 12345234
+
+      - blockDeviceName: pool-1-bd-4
+        id: pool-1-bd4
+        // System generated
+        capacity: 12345234
+
+  - name: gke-cstor-it-default-pool-2
+
+    groups:
+    - groupName: group1
+      blockDevices:
+      - blockDeviceName: pool-2-bd-1
+        // System generated
+        id: pool-2-bd1
+        // System generated
+        capacity: 12345234
+      - blockDeviceName: pool-2-bd-2
+        // System generated
+        id: pool-2-bd2
+        // System generated
+        capacity: 12345234
+
+    // New block device group added
+    - groupName: group2
+      blockDevices:
+      - blockDeviceName: pool-2-bd-3
+        // System generated
+        id: pool-2-bd3
+        // System generated
+        capacity: 12345234
+
+      - blockDeviceName: pool-2-bd-4
+        // System generated
+        id: pool-2-bd4
+        // System generated
+        capacity: 12345234
+
+```
+
+**NOTE:**
+In case of striped pool expansion no new group but block device in the present
+group as striped has only one groups but in all other raid groups a new group is
+added for expansion.
+
+### Operations Workflow
+( Includes Pool Expansion, Pool Deletion, Block Device Offline, Block Device Replacment and Pool Migration)
+
+Work done by CStor-operator after the user step is performed
+
+1. The node spec hash will change as a result of user adding/removing
+   blockDevices to CSPC or changing hostName for pool and then figure out what
+   operation it is!
+
+2. Host name changed in the CSPC can cause following:( Pool Migration)
+   Cstor-Operator will patch the corresponding CSP with the host name in the CSP
+   spec. Pool-mgmt should handle this change in CSP
+
+3. Block device removal can cause following:
+   i) Top level vdev not lost(Block Device Offline):
```

Comment:
We are introducing the concepts of vdev & top level vdev. Does it make sense to explain these concepts in this doc?

Reply:
cc @vishnuitta